x-pack/plugins/security/public/views/management/edit_role/lib/transform_role_for_save.ts

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { IndexPrivilege, Role } from '../../../../../common/model';
+import { isGlobalPrivilegeDefinition } from '../../../../lib/privilege_utils';
+
+export function transformRoleForSave(role: Role, spacesEnabled: boolean) {
```

Comment:
> However, transformRoleForSave maintains the Role interface while saveRole does not, was that the intent with the separation of these two?

I'd like to give myself that much credit, but that wasn't the case. We had the `role.elasticsearch.*` transforms within a React component before, and when I needed to add the `role.kibana.*` transforms, I decided to pull that logic out of the React component and consolidate it here. At the time, I didn't realize/forgot that we also had the other function which stripped out those other fields.

How do you feel about including the `omit` logic from https://github.com/elastic/kibana/blob/master/x-pack/plugins/security/public/objects/lib/roles.ts#L12 here, and having this function called from there instead of the react component? I'm ok updating this so that it no longer maintains the `Role` interface.

Reply:
Your proposal sounds perfectly fine to me!